runtime/src/accounts_db.rs

```diff
@@ -882,15 +882,21 @@ impl AccountsDB {
         accounts: &[(&Pubkey, &Account)],
     ) -> (Vec<(Slot, AccountInfo)>, u64) {
         let mut reclaims: Vec<(Slot, AccountInfo)> = Vec::with_capacity(infos.len() * 2);
-        let mut inserts = vec![];
         let index = self.accounts_index.read().unwrap();
         let mut update_index_work = Measure::start("update_index_work");
-        for (info, pubkey_account) in infos.into_iter().zip(accounts.iter()) {
-            let pubkey = pubkey_account.0;
-            if let Some(info) = index.update(slot_id, pubkey, info, &mut reclaims) {
-                inserts.push((pubkey, info));
-            }
-        }
+        let inserts: Vec<_> = infos
+            .into_iter()
+            .zip(accounts.iter())
+            .filter_map(|(info, pubkey_account)| {
+                let pubkey = pubkey_account.0;
+                if let Some(info) = index.update(slot_id, pubkey, info, &mut reclaims) {
```

Comment:
btw, you can write this as:
`index.update(slot_id, pubkey, info, &mut reclaims).map(|info| (pubkey, info))`

Reply:
heh, duh, thanks!